### PR TITLE
moveEmplace of static array to not try and destruct garbage

### PR DESCRIPTION
**Pull request by radcapricorn**
Per discussion in https://github.com/dlang/druntime/pull/3638#discussion_r766310457:

Target is assumed uninitialzed, `move` would issue a destructor call if `T` has a destructor, thereby destructing garbage. This should be a recursive `moveEmplace` instead of `move`.

### CODE REVIEW

**Review by thewilsonator — APPROVED** [review]

**Comment by thewilsonator** [comment]
should this target stable?

**Review by RazvanN7 — APPROVED** [review]

**Comment by RazvanN7** [comment]
I think master is fine.

File: src/core/lifetime.d [hunk 1]
```diff
@@ -2154,7 +2154,7 @@ private void moveEmplaceImpl(T)(scope ref T target, return scope ref T source)
     else static if (__traits(isStaticArray, T))
     {
         for (size_t i = 0; i < source.length; ++i)
```
Comment (nordlow) [inline src/core/lifetime.d:2156]: Shouldn't this have a special casing that resorts to a `memcpy` when `target` is a type lacking copy-constructor nor postblit?

**Comment by nordlow** [comment]
I believe `emplace` should be a compiler builtin just as `new` is.

**Comment by radcapricorn** [comment]
> Btw, I believe `emplace` should be a compiler builtin perhaps by copying C++'s emplacement-new [1]
> 
> [1] https://en.cppreference.com/w/cpp/language/new#Placement_new.

Yeah, me too: https://forum.dlang.org/thread/jraxclewuzrqvtrguhct@forum.dlang.org
Feel free to join to the discussion ;)

**Pull request merged**